python/CMakeLists.txt

```diff
@@ -1,38 +1,34 @@
-set(Python_ADDITIONAL_VERSIONS 3.7 3.6 3.5)  # adjust with CMake minimum FindPythonInterp
-find_package(PythonLibsNew 2.7 REQUIRED)
 find_package(SWIG REQUIRED)
+INCLUDE(UseSWIG)
+
+find_package(PythonInterp 2.7 REQUIRED)
+find_package(PythonLibs 2.7 REQUIRED)
 find_package(NumPy REQUIRED)
 
 set(XCFUN_SWIG ${CMAKE_CURRENT_LIST_DIR}/xcfun_swig.i)
-set(XCFUN_SWIG_WRAP "xcfun_swigPYTHON_wrap.c")
 
-add_custom_command(
-  OUTPUT ${XCFUN_SWIG_WRAP}
-  COMMAND "${SWIG_EXECUTABLE}"
-  ARGS "-python" "-I${PROJECT_SOURCE_DIR}/include" "-outdir" "${CMAKE_CURRENT_BINARY_DIR}" "-o" ${XCFUN_SWIG_WRAP} ${XCFUN_SWIG}
-  WORKING_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}
-  DEPENDS XCFun ${XCFUN_SWIG}
-  )
+set_property(SOURCE "${XCFUN_SWIG}" PROPERTY SWIG_FLAGS "-I${CMAKE_BINARY_DIR}/include" 
+                                                        "-I${PROJECT_SOURCE_DIR}/api")
+set_property(SOURCE "${XCFUN_SWIG}" PROPERTY SWIG_OUTDIR ${CMAKE_CURRENT_BINARY_DIR})
+
+# requires CMake 3.8
```

Comment:
I'm not against moving to CMake 3.8, but that probably requires larger consensus.

Reply:
For me it is OK if the bindings need 3.8 but the configuration should not die if CMake is below 3.8 for those who configure without the bindings.

Reply:
The current code works with CMake below 3.5, the version that requires CMake 3.8 is commented out. Lets leave it like this and change the code to commented version once a newer CMake is required anyways.